isthmus/src/main/java/io/substrait/isthmus/SubstraitRelVisitor.java

```diff
@@ -196,6 +197,19 @@ public Rel visit(LogicalJoin join) {
 
   @Override
   public Rel visit(LogicalCorrelate correlate) {
+    // left input of correlated-join is similar to the left input of a logical join
+    apply(correlate.getLeft());
+
+    // right input of correlated-join is similar to a correlated sub-query
+    apply(correlate.getRight());
+
+    var joinType =
```

Comment:
Is `joinType` an unused variable?

Reply:
Yes, I believe leaving this unused-variable in the code for now helps in code readability. I mentioned the following in response to another comment ([here](https://github.com/substrait-io/substrait-java/pull/106#discussion_r1032851122)): 
> I have not removed the join-type variable because it helps in clarifying the expected types till the next PR is merged.

Given this features is developed in iterations, I believe it is best to not remove this variable as it will be needed in the next PR. It is not dead code.

Reply:
Make sense to me.  

If I understand correctly, the `switch` is to check if `JoinType` is indeed Inner or Left. For other types, something is wrong in the passed in `LogicalCorrelate`.  Does it make sense to use `UnsupportedOperationException` here, or maybe use `assert` to make sure the condition is met?

Reply:
I think this is a good point. In case of `LogicanlCorrelate` and other join type is invalid. How about throwing `IllegalArgumentException` if any other join type is passed?

Reply:
Yes, `IllegalArgumentException` makes sense for invalid input.